src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,116 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
             execProfilesName = projectName;
         }
         if (validateForBuild) {
-            validateForBuild(jetHomeObj);
+            validateForBuild(excelsiorJet);
         }
 
-        return jetHomeObj;
+        processDependencies();
     }
 
-    private void validateForBuild(JetHome jetHomeObj) throws JetTaskFailureException {
+    public void processDependencies() throws JetTaskFailureException {
+        for (DependencySettings dependency : dependencies) {
+            if (!dependency.isValid()) {
+                throw new JetTaskFailureException(s("JetApi.InvalidDependencySetting", dependency.idStr()));
+            }
+        }
+        List<DependencySettings> dependenciesSettings = dependencies.stream()
+                .filter(DependencySettings::isSettings)
```

Comment:
Unfortunately, these methods are not mutually exclusive, but reflect objective reality. This is due to the existing design of DependencySetting (the mixing of settings and external dependencies concepts in the plugins), whereby the object can be both setting and external dependency simultaneously. Or be neither one nor the other.

The only thing that will ensure that the object is either setting or external dependency - is checks in the code above, but they can break.

I expect that the most common mistake is to add id to the external dependency. For these reasons all the objects with path must be regarded as external dependencies. But if a user accidentally add an existing path to the settings, then these settings will not be applied, which may lead to the disabling of the protection of the user. What out of this is worse? What predicate should I keep?